BinaryObjectScanner/Protection/DigiGuard.cs

```diff
@@ -0,0 +1,95 @@
+    /// Second list of archived GLFC press releases (Search for "pr_"): https://web.archive.org/web/*/http://glfc.com/* 
+    /// News article regarding GLFC and Accolade Games: https://www.gamespot.com/articles/accolade-games-on-dvd/1100-2460436/
+    /// eBay listing for the "BIGWIG SOFTWARE LOCKER", which very likely contains DigiGuard: https://www.ebay.com/itm/325573968970
+    /// </summary>
+    public class DigiGuard : IPathCheck, IPortableExecutableCheck
+    {
+        /// <inheritdoc/>
+#if NET48
+        public string CheckPortableExecutable(string file, PortableExecutable pex, bool includeDebug)
+#else
+        public string? CheckPortableExecutable(string file, PortableExecutable pex, bool includeDebug)
+#endif
+        {
+            // Get the sections from the executable, if possible
+            var sections = pex.Model.SectionTable;
+            if (sections == null)
+                return null;
+
+            // Found in "Start.exe" in IA item "Nova_DellBigWIGDVD_USA"/Redump entry 108588.
+            var name = pex.FileDescription;
+            if (name?.Equals("DigiGuard3 Client") == true)
+                return $"DigiGuard3";
+
+            // Found in "Start.exe" in IA item "Nova_DellBigWIGDVD_USA"/Redump entry 108588.
+            name = pex.LegalTrademarks;
+            if (name?.Equals("DigiGuard") == true)
+                return $"DigiGuard";
+
+            // Found in "PJS3.exe" in IA item "Nova_DellBigWIGDVD_USA"/Redump entry 108588.
+            name = pex.ProductName;
+            if (name?.Equals("Greenleaf Wrapper3") == true)
+                return $"DigiGuard";
+
+            return null;
+        }
+
+        /// <inheritdoc/>
+#if NET48
+        public ConcurrentQueue<string> CheckDirectoryPath(string path, IEnumerable<string> files)
+#else
+        public ConcurrentQueue<string> CheckDirectoryPath(string path, IEnumerable<string>? files)
+#endif
+        {
+            var matchers = new List<PathMatchSet>
+            {
+                // Found in IA item "Nova_DellBigWIGDVD_USA"/Redump entry 108588.
+                new PathMatchSet(new PathMatch("DecryptWrap.exe", useEndsWith: true), "DigiGuard"),
+
+                // There are at least two additional specifically named DecryptWrap files, "DecryptWrapTW2000.exe" and "DecryptWrapTW2KCode.exe" in IA item "Nova_DellBigWIGDVD_USA"/Redump entry 108588.
+            };
+
+            return MatchUtil.GetAllMatches(files ?? System.Array.Empty<string>(), matchers, any: true);
+        }
+
+        /// <inheritdoc/>
+#if NET48
+        public string CheckFilePath(string path)
+#else
+        public string? CheckFilePath(string path)
+#endif
+        {
+            var matchers = new List<PathMatchSet>
+            {
+                // Found in IA item "Nova_DellBigWIGDVD_USA"/Redump entry 108588.
+                new PathMatchSet(new PathMatch("DecryptWrap.exe", useEndsWith: true), "DigiGuard"),
```

Comment:
I think I had made a "FilePathMatch" or something similar that you can use. That will ensure that only the filename is caught.